zip-0311.rst

```diff
@@ -2,7 +2,441 @@
+CarsRFast's transparent address. They can see a payment has been made, but claim that they
+have not received a payment from Edward. He can create a payment disclosure that reveals:
+
+- Edward was the sender of the given transaction.
+
+and does not reveal:
+
+- Edward's payment address.
+- Any of Edward's other diversified payment addresses.
+- Any of Edward's other transactions.
+
+Shielded withdrawals from an exchange
+-------------------------------------
+
+CorpBux is an exchange with a transparent hot wallet, and enables customers to withdraw
+their funds to Sapling addresses. They can create a payment disclosure, that they can give
+to their customers, that reveals:
+
+- The correct amount was sent to the customer's shielded recipient address in the given
+  transaction.
+- CorpBux was the sender of the given transaction.
+
+
+Requirements
+============
+
+- The payment disclosure may disclose the contents of zero or more Sapling outputs within
+  a single transaction. (Zero outputs is useful for z2t transactions.)
+- Payment disclosures can only be created by a sender of the transaction,
+  and are non-malleable (that is, no-one else can take one or more payment
+  disclosures and construct a new one that they would not have been able
+  to make independently).
+- A payment disclosure may be tied to a challenge, creating an interactive proof.
+- Senders with a Sapling address may choose to provably reveal that address as part of the
+  payment disclosure.
+- Senders are not required to remember any ephemeral data from the creation of a transaction
+  in order to produce a payment disclosure for that transaction. (Performance may be
+  improved if they do cache witness data for spent notes.)
+
+
+Conventions
+===========
+
+The following functions used in this ZIP are defined in the Zcash protocol specification:
+[#protocol]_
+
+- :math:`\mathsf{DiversifyHash}(\mathsf{d})` [#protocol-concretediversifyhash]_
+
+- :math:`\mathsf{SpendAuthSig.RandomizePrivate}(α, \mathsf{sk})`,
+  :math:`\mathsf{SpendAuthSig.Sign}(\mathsf{sk}, m)`, and
+  :math:`\mathsf{SpendAuthSig.Verify}(\mathsf{vk}, m, σ)` [#protocol-concretespendauthsig]_
+
+We reproduce some notation and functions from [#protocol]_ here for convenience:
+
+- :math:`[k] P` means scalar multiplication of the elliptic curve point :math:`P` by the
+  scalar :math:`k`.
+
+- :math:`\mathsf{BLAKE2b}\text{-}\mathsf{256}(p, x)` refers to unkeyed BLAKE2b-256 in
+  sequential mode, with an output digest length of 32 bytes, 16-byte personalization
+  string :math:`p`, and input :math:`x`.
+
+We also define the following notation here:
+
+- :math:`[a..b]` means the sequence of values inclusive of :math:`a` and exclusive of
+  :math:`b`.
+
+- :math:`\mathsf{length}(a)` means the length of the sequence :math:`a`.
+
+
+Specification
+=============
+
+Payment disclosure data structure
+---------------------------------
+
+A payment disclosure has the following fields:
```

Comment:
Let's reorder this section to have the same ordering as the [v5 tx format](https://zips.z.cash/zip-0225#transaction-format) (transparent, Sapling, Orchard).